pkg/services/apikey/apikeyimpl/apikey.go

```diff
@@ -6,23 +6,42 @@ import (
 	"github.com/grafana/grafana/pkg/infra/db"
 	"github.com/grafana/grafana/pkg/services/apikey"
 	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/services/quota"
 	"github.com/grafana/grafana/pkg/setting"
 )
 
 type Service struct {
 	store store
 }
 
-func ProvideService(db db.DB, cfg *setting.Cfg) apikey.Service {
+func ProvideService(db db.DB, cfg *setting.Cfg, quotaService quota.Service) (apikey.Service, error) {
+	s := &Service{}
 	if cfg.IsFeatureToggleEnabled(featuremgmt.FlagNewDBLibrary) {
-		return &Service{
-			store: &sqlxStore{
-				sess: db.GetSqlxSession(),
-				cfg:  cfg,
-			},
+		s.store = &sqlxStore{
+			sess: db.GetSqlxSession(),
+			cfg:  cfg,
 		}
 	}
-	return &Service{store: &sqlStore{db: db, cfg: cfg}}
+	s.store = &sqlStore{db: db, cfg: cfg}
+
+	defaultLimits, err := readQuotaConfig(cfg)
+	if err != nil {
+		return s, err
+	}
+
+	if err := quotaService.AddReporter(context.TODO(), &quota.NewQuotaReporter{
```

Comment:
@papagian could you elaborate a bit on the `context.TODO()` i do not usually use this. Only the `context.Background()`.

Curious to understand the TODO use case here

Reply:
I don't think it makes difference given that:
https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/context/context.go;l=199

I have used `TODO()` since `it is not yet available` given that this code is executed during the service initialisation. Maybe it does not make much sense for `quotaService.AddReporter()` to expect a context (since it is always called during a service initialisation) so I can better remove it.